.github/workflows/ci-go.yml

```
@@ -0,0 +1,69 @@
+# Continuous integration handling for Go
+name: ci-go
+
+on:
+  pull_request:
+    paths:
+      - .github/workflows/test.yml
+      - .golangci.yml
+      - .go-version
+      - go.mod
+      - '**.go'
+
+permissions:
+  contents: read
+
+jobs:
+  golangci-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - id: go-version
+        # Reference: https://github.com/actions/setup-go/issues/23
+        run: echo "::set-output name=version::$(cat ./.go-version)"
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ steps.go-version.outputs.version }}
+      - run: go mod download
+      - uses: golangci/golangci-lint-action@v2
+        with:
+          skip-go-installation: true
+  terraform-provider-corner:
+    defaults:
+      run:
+        working-directory: terraform-provider-corner
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/checkout@v2
```

Completely optional, but seeing two back-to-back `actions/checkout@v2` calls could potentially be jarring to someone new modifying this in the future.
```suggestion
      # Checkout current repo
      - uses: actions/checkout@v2
      # Checkout terraform-provider-corner repo
      - uses: actions/checkout@v2
```